states/kv/utils.go

```diff
@@ -0,0 +1,14 @@
+package kv
+
+import (
+	"path"
+	"strings"
+)
+
+func joinPath(parts ...string) string {
+	r := path.Join(parts...)
+	if len(parts) > 0 && strings.HasSuffix(parts[len(parts)-1], "/") {
```

Comment:
the second condition could be simple:
`strings.HasSuffix(r, "/")`

Reply:
`r` will never has "/" suffix since path.Join drops the tailing slash